documentation/src/main/asciidoc/migration/index.asciidoc

```diff
@@ -0,0 +1,3351 @@
+
+SearchResult<Book> result = searchSession.search( Book.class )
+        .where( f -> f.matchAll() )
+        .aggregation( countByGenreKey, f -> f.terms()
+                .field( "genre", Genre.class )
+                .orderByCountDescending()
+                .minDocumentCount( 1 )
+                .maxTermCount( 3 ) )
+        .fetch( 20 );
+
+List<Book> hits = result.hits();
+
+Map<Genre, Long> countByGenre = result.aggregation( countByGenreKey );
+----
+
+See [this section of the documentation] for more information.
+
+===== Range faceting
+
+Let's consider the query below:
+
+[source, JAVA]
+----
+FullTextEntityManager fullTextEntityManager = Search.getFullTextEntityManager( em );
+QueryBuilder qb = fullTextEntityManager.getSearchFactory()
+        .buildQueryBuilder().forEntity( Book.class ).get();
+
+FacetingRequest priceFacetingRequest = qb.facet()
+        .name( "priceFaceting" )
+        .onField( "price" )
+        .range()
+        .below( 1000 ).excludeLimit()
+        .from( 1001 ).to( 1500 ).excludeLimit()
+        .above( 1500 )
+        .orderedBy( FacetSortOrder.COUNT_DESC )
+        .includeZeroCounts( false )
+        .maxFacetCount( 3 )
+        .createFacetingRequest();
+
+FullTextQuery fullTextQuery = fullTextEntityManager.createFullTextQuery( qb.all().createQuery(), Book.class );
+fullTextQuery.setMaxResults( 20 );
+
+FacetManager facetManager = fullTextQuery.getFacetManager();
+facetManager.enableFaceting( priceFacetingRequest );
+
+List hits = fullTextQuery.getResultList();
+
+List<Facet> facets = facetManager.getFacets( "priceFaceting" );
+----
+
+It would look like this in Hibernate Search 6:
+
+[source, JAVA]
+----
+SearchSession session = Search.session( entityManager );
+
+AggregationKey<Map<Range<Double>, Long>> countByPriceRangeKey = AggregationKey.of( "countByPriceRange" );
+
+SearchResult<Book> result = searchSession.search( Book.class )
+        .where( f -> f.matchAll() )
+        .aggregation( countByPriceRangeKey, f -> f.range()
+                .field( "price", Double.class )
+                .range( Range.lessThan( 1000.0 ) )
+                .range( Range.canonical( 1000.0, 1500.0 ) )
+                .range( Range.atLeast( 1500.0 ) )
+                // Not equivalent to 'orderedBy'
+                // Not equivalent to 'includeZeroCounts'
+                // Not equivalent to 'maxFacetCount'
+        .fetch( 20 );
+
+List<Book> hits = result.hits();
+
+Map<Range<Double>, Long> countByPriceRange = result.aggregation( countByPriceRangeKey );
+----
+
+[NOTE]
+====
+`orderedBy`, `includeZeroCounts` and `maxFacetCount` have no equivalent Hibernate Search 6 range aggregations:
+all given ranges will always be included in the resulting `Map`.
+
+The behavior of these methods can be implemented by post-processing the `Map` in user code.
+====
+
+[[searching-facets-selectfacets]]
+===== Drill-down with `.selectFacets`
+
+In Hibernate Search 5, the `.selectFacets` method used to allow "drill-down",
```

Comment:
s/used/was used/ ?

Reply:
"used to" as in "in the past we allowed the drill-down operation, but now we don't allow it anymore".